### PR TITLE
First commits

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request contains the first commits for an Ansible role to install the CrowdStrike Falcon sensor.

## 💭 Motivation and context ##

We must install the CrowdStrike Falcon sensor to maintain the COOL ATO.

## 🧪 Testing ##

All automated tests pass.  I was also able to use these changes to build a new OpenVPN AMI for our staging COOL as part of the testing for cisagov/openvpn-packer#87.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,16 +3,9 @@
...
-[cisagov](https://github.com/cisagov) Ansible role GitHub project
-started, where that Ansible role requires an AWS test user for
-automated testing.
-
-This skeleton project contains [licensing information](LICENSE), as
-well as [pre-commit hooks](https://pre-commit.com) and [GitHub
-Actions](https://github.com/features/actions) configurations
-appropriate for an Ansible role, as well as the Terraform code to
-create the AWS test user.
+This is an Ansible role for installing [CrowdStrike
```
Comment (mcdonnnj) [inline README.md]: This aligns with the language CrowdStrike themselves use in the [installation guide](https://www.crowdstrike.com/blog/tech-center/install-falcon-sensor-for-linux/).

```suggestion
This is an Ansible role for installing the [CrowdStrike
```

File: defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+# The directory where CrowdStrike Falcon sensor is installed
```
Comment (mcdonnnj) [inline defaults/main.yml]:
```suggestion
# The directory where the CrowdStrike Falcon sensor is installed
```

File: README.md [hunk 3]
```diff
@@ -69,18 +62,15 @@ None.
...
 
-None.
-
-<!--
 | Variable | Description | Default | Required |
 |----------|-------------|---------|----------|
-| optional_variable | Describe its purpose. | `default_value` | No |
-| required_variable | Describe its purpose. | n/a | Yes |
--->
+| install_directory | The directory where CrowdStrike Falcon sensor is installed. | `/opt/CrowdStrike` | No |
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
| install_directory | The directory where the CrowdStrike Falcon sensor is installed. | `/opt/CrowdStrike` | No |
```

File: meta/main.yml [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
+  # https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#using-roles-at-the-play-level
+  - name: cdm_certificates
+    src: https://github.com/cisagov/ansible-role-cdm-certificates
+
 galaxy_info:
-  author: First Last
+  author: Shane Frasier
   company: CISA Cyber Assessments
-  description: Skeleton Ansible role with test user
+  description: Install CrowdStrike Falcon sensor for CISA CDM
```
Comment (mcdonnnj) [inline meta/main.yml]:
```suggestion
  description: Install the CrowdStrike Falcon sensor for CISA CDM
```

File: molecule/default/tests/test_default.py [hunk 5]
```diff
@@ -4,15 +4,24 @@
...
     os.environ["MOLECULE_INVENTORY_FILE"]
 ).get_hosts("all")
 
 
-@pytest.mark.parametrize("x", [True])
-def test_packages(host, x):
-    """Run a dummy test, just to show what one would look like."""
-    assert x
+def test_falcon_sensor_installed(host):
+    """Test that CrowdStrike Falcon sensor was installed."""
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
    """Test that the CrowdStrike Falcon sensor was installed."""
```

File: molecule/default/tests/test_default.py [hunk 6]
```diff
@@ -4,15 +4,24 @@
...
+    dir_full_path = "/opt/CrowdStrike"
+    directory = host.file(dir_full_path)
+    assert directory.exists
+    assert directory.is_directory
+    # Make sure that the directory is not empty
+    assert host.run_expect([0], f'[ -n "$(ls -A {dir_full_path})" ]')
+
+
+def test_falcon_sensor_enabled(host):
+    """Test that CrowdStrike Falcon sensor is enabled."""
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
    """Test that the CrowdStrike Falcon sensor is enabled."""
```

File: tasks/Debian.yml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Install CrowdStrike Falcon sensor (Debian)
```
Comment (mcdonnnj) [inline tasks/Debian.yml]:
```suggestion
- name: Install the CrowdStrike Falcon sensor (Debian)
```

File: tasks/main.yml [hunk 8]
```diff
@@ -1,2 +1,77 @@
...
+    params:
+      files:
+        - "{{ ansible_distribution }}_{{ ansible_distribution_release }}.yml"
+        - "{{ ansible_distribution }}.yml"
+        - "{{ ansible_os_family }}.yml"
+      paths:
+        - "{{ role_path }}/vars"
+
+# Check if CrowdStrike Falcon sensor is already installed
+- name: Check if CrowdStrike Falcon sensor is already installed
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
- name: Check if the CrowdStrike Falcon sensor is already installed
```

File: tasks/main.yml [hunk 9]
```diff
@@ -1,2 +1,77 @@
...
+  ansible.builtin.stat:
+    path: "{{ install_directory }}"
+  register: falcon_directory
+
+- name: Install CrowdStrike Falcon sensor
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
- name: Install the CrowdStrike Falcon sensor
```

File: tasks/main.yml [hunk 10]
```diff
@@ -1,2 +1,77 @@
...
+  block:
+    - name: Grab CrowdStrike Falcon sensor system package from S3
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
    - name: Grab the CrowdStrike Falcon sensor system package from S3
```

File: tasks/main.yml [hunk 11]
```diff
@@ -1,2 +1,77 @@
...
+            - "{{ role_path }}/tasks"
+
+    - name: Enable the CrowdStrike Falcon sensor service
+      ansible.builtin.service:
+        enabled: yes
+        name: falcon-sensor
+        # Required to set the update_hostname option below
+        # state: started
+
+    # - name: Configure the CrowdStrike Falcon sensor to send hostname updates
```
Comment (mcdonnnj) [inline tasks/main.yml]: Comment explaining why this is here but disabled?

File: tasks/main.yml [hunk 12]
```diff
@@ -1,2 +1,77 @@
...
+    #     argv:
+    #       - "{{ install_directory }}/sbin/nessuscli"
+    #       - fix
+    #       - --set
+    #       - update_hostname=yes
+    #   changed_when: nessuscli_output.rc == 0
+    #   failed_when: nessuscli_output.rc != 0
+    #   register: nessuscli_output
+
+    - name: Delete local copy of CrowdStrike Falcon sensor system package
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
    - name: Delete local copy of the CrowdStrike Falcon sensor system package
```

File: tasks/main.yml [hunk 13]
```diff
@@ -1,2 +1,77 @@
...
+      ansible.builtin.file:
+        path: /tmp/{{ package_object_name }}
+        state: absent
+      become: no
+      delegate_to: localhost
+
+    - name: Delete remote copy of CrowdStrike Falcon sensor system package
```
Comment (mcdonnnj) [inline tasks/main.yml]:
```suggestion
    - name: Delete remote copy of the CrowdStrike Falcon sensor system package
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 11 above)
Comment (jsf9k) [reply tasks/main.yml]: Removed in commit 3a01153.

**Review by dav3r — APPROVED** [review]
Looks great! 💪 💼

**Review by dav3r — APPROVED** [review]
👍 👍

File: meta/main.yml
Comment (mcdonnnj) [inline meta/main.yml:1]: Please update the supported platforms to reflect what molecule platforms are still enabled.

File: tasks/main.yml [hunk 14]
```diff
@@ -1,2 +1,64 @@
...
+        state: absent
+      become: no
+      delegate_to: localhost
+
+    - name: Delete remote copy of the CrowdStrike Falcon sensor system package
+      ansible.builtin.file:
+        path: /tmp/{{ package_object_name }}
+        state: absent
+
+  when: not falcon_directory.stat.exists
```
Comment (mcdonnnj) [inline tasks/main.yml]: Something about the extra line bothers me. I'm fine if you decide against it but figured I'd mention it at least.

```suggestion
  when: not falcon_directory.stat.exists
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
One change request and one thought for your consideration.

File: README.md [hunk 15]
```diff
@@ -69,18 +62,15 @@ None.
...
-| optional_variable | Describe its purpose. | `default_value` | No |
-| required_variable | Describe its purpose. | n/a | Yes |
--->
+| install_directory | The directory where the CrowdStrike Falcon sensor is installed. | `/opt/CrowdStrike` | No |
+| third_party_bucket_name | The name of the AWS S3 bucket where third-party software is located. | `cisa-cool-third-party-production` | No |
 
 ## Dependencies ##
 
-None.
+- [cisagov/ansible-role-cdm-certificates](https://github.com/cisagov/ansible-role-cdm-certificates)
```
Comment (mcdonnnj) [inline README.md:72]: There are not in the role metadata. Also does https://github.com/cisagov/openvpn-packer/commit/55c128035e40f7322decd1516825ad1e6704fbba impact this configuration?

File: README.md [hunk 16]
```diff
@@ -69,18 +62,15 @@ None.
...
+- [cisagov/ansible-role-dhs-certificates](https://github.com/cisagov/ansible-role-dhs-certificates)
```
Comment (mcdonnnj) [inline README.md]: There are not in the role metadata. Also does https://github.com/cisagov/openvpn-packer/commit/55c128035e40f7322decd1516825ad1e6704fbba impact this configuration?

Comment (jsf9k) [reply README.md]: Please see commit 3508fdb.

File: meta/main.yml
Comment (jsf9k) [reply meta/main.yml:1]: Please see commit 7d95f5b.

**Review by mcdonnnj — APPROVED** [review]
:shipit:

**Pull request merged**